smoke_test.sh

```diff
@@ -0,0 +1,90 @@
+# Spin up iqe pod and execute IQE tests in it
+
+# Env vars defined by caller:
+#IQE_PLUGINS="plugin1,plugin2" -- pytest plugins to run separated by "," #IQE_MARKER_EXPRESSION="mymarker" -- pytest marker expression
+#IQE_FILTER_EXPRESSION="something AND something_else" -- pytest filter, can be "" if no filter desired
+#NAMESPACE="mynamespace" -- namespace to deploy iqe pod into, can be set by 'deploy_ephemeral_env.sh'
+
+
+function kill_port_fwd {
+    echo "Caught signal, kill port forward"
+    if [ ! -z "$PORT_FORWARD_PID" ]; then kill $PORT_FORWARD_PID; fi
+}
+
+# The CJI var name will need to be exported in the main pr_check.sh
+oc apply -n $NAMESPACE -f $APP_ROOT/$CJI_PATH
+
+job_name=$APP_NAME-smoke-tests-iqe
+found=false
+end=$((SECONDS+60))
+
+echo "Waiting for Job $job_name to appear"
+
+while [ $SECONDS -lt $end ]; do
+    if `oc get job $job_name -n $NAMESPACE >/dev/null 2>&1`; then
+        found=true
+        break
+    fi
+    sleep 1
+done
+
+if [ "$found" == "false" ] ; then
+    echo "Job $job_name failed to appear"
+    exit 1
+fi
+
+echo "Waiting for Job $job_name to be running"
+running=false
+pod=""
+
+# The jq magic will find all running pods in the ns and regex on the app name
+# Loop over for SECONDS and send back the pod's name once found
+while [ $SECONDS -lt $end ]; do
+    pod=$(oc get pods -n $NAMESPACE -o json | jq -r --arg JOB $job_name '.items[] | select(.status.phase=="Running") | select(.metadata.name|test($JOB)) .metadata.name')
+    if [[ -n $pod ]]; then
+        running=true
+        break
+    fi
+    sleep 5
+done
+
+if [ "$running" == "false" ] ; then
+    echo "Job $job_name failed to start"
+    exit 1
+fi
+
+# Pipe logs to background to keep them rolling in jenkins
+oc logs -n $NAMESPACE $pod -f &
+
+# Wait for the job to Complete or Fail before we try to grab artifacts
+# condition=complete does trigger when the job fails
+oc wait --timeout=3m --for=condition=Complete -n $NAMESPACE job/$job_name 
```

Comment:
Some apps' tests will definitely take longer than 3 min... should we make timeout configurable via some env var defined at the top of pr_check.sh and default to something kind of large like 30min to start with?